include/exadg/poisson/spatial_discretization/operator.cpp

```diff
@@ -163,6 +163,9 @@ Operator<dim, n_components, Number>::distribute_dofs()
 
       dealii::DoFTools::make_periodicity_constraints<dim, dim, Number>(periodic_faces_dof,
                                                                        affine_constraints);
+
+      dealii::DoFTools::make_periodicity_constraints<dim, dim, Number>(
+        periodic_faces_dof, affine_constraints_periodicity_and_hanging_nodes);
     }
```

Comment:
The above two functions are potentially expensive. I recommend to only call the above functions for `affine_constraints_periodicity_and_hanging_nodes` above, then call `close()` on that object, and finally call
```c++
  affine_constraints.copy_from(affine_constraints_periodicity_and_hanging_nodes);
```
before going on with the code below. I think that also documents more clearly that the regular constraints are an extended variant, and always have the same baseline constraints. (This would avoid the need for the commend you add in line 193 in my opinion.)

Reply:
see https://github.com/exadg/exadg/pull/506/commits/db18a5ec73c5460a1bc5f524ef5f7c86a9570738

Reply:
I think it makes sense to introduce a utility function and be more strict with "don't repeat yourself" even if the code is just a few calls to dealii utilities ... see https://github.com/exadg/exadg/pull/506/commits/d2d07b9f51f07a1fae918baddc1ed4de014097d9.